docs/typescript/reference/configuration.md

```diff
@@ -1,186 +1,126 @@
+Additionally, the DBOS library will fall back to `dbos-config.yaml` if no configuration object is provided.
 
-```yaml
-http:
-  cors_middleware: true
-  credentials: true
-  allowed_origins:
-    - 'https://partner.com'
-    - 'https://app.internal.com'
-```
----
-
-### Application
+Here is an example configuration file with default parameters:
 
-Applications can optionally use the application configuration to define custom properties as key-value pairs.
-These properties can be retrieved from any [context](./transactapi/oldapi/contexts) via the [`getConfig`](./transactapi/oldapi/contexts#ctxtgetconfig) method.
-
-**Example**:
-```yaml
-application:
-  PAYMENTS_SERVICE: 'https://stripe.com/payment'
+```shell
+name: my-app
+language: node
+database_url: ${DBOS_DATABASE_URL}
+runtimeConfig:
+  start:
+    - node dist/main.js
 ```
 
-### Environment Variables
-
-Applications can optionally use the `env` configuration to define environment variables.
-These are set in your application before its code is initialized and can be retrieved from `process.env` like any other environment variables.
-For example, the `WEB_PORTAL` variable set below could be retrieved from an application as `process.env.WEB_PORTAL`.
-Environment variables configured here are automatically exported to and set in DBOS Cloud.
+### Configuration File Fields
 
-**Example**:
-```yaml
-env:
-  WEB_PORTAL: 'https://example.com'
-```
+::::info
+You can use environment variables for configuration values through the syntax `field: ${VALUE}`.
+::::
 
----
+Each `dbos-config.yaml` file has the following fields and sections:
 
-### Telemetry
+- **name**: Your application's name. Must match the name supplied to the DBOS constructor.
+- **language**: The application language. Must be set to `node` for TypeScript applications.
+- **database_url**: A connection string to a Postgres database. This connection string is used by tools such as the [DBOS CLI](./tools/cli.md) and [DBOS debugger](../tutorials/debugging.md). It has the same format as (and should match) the connection string you pass to the DBOS constructor.
+- **database**: The [database section](#database-section).
+- **runtimeConfig**: The [runtime section](#runtime-section).
 
-You can use the configuration file to tune the behavior of DBOS logging facility.
-Note all options in this section are optional and will, if not specified, use the default values indicated in the example below.
+#### Database Section
 
-#### Logs
-- **logLevel**: Filters, by severity, what logs should be printed. Defaults to `'info'`. Using [`npx dbos start -l <logLevel>`](./tools/cli#npx-dbos-start) overrides this config parameter.
-- **addContextMetadata**: Enables the addition of contextual information, such as workflow identity UUID, to each log entry. Defaults to `true`.
-- **silent**: Silences the logger. Defaults to `false`.
+- **migrate**: A list of commands to run to apply your application's schema to the database. 
```

Comment:
We shouldn't advertise this anymore

Reply:
We'll keep it as long as the cloud uses it